### PR TITLE
[Model Monitoring] Update model endpoint's current stats and drift measures in the writer

### PR DESCRIPTION
**Pull request by jond01**
Resolves [ML-5767](https://iguazio.atlassian.net/browse/ML-5767).

It is tested with a unit test and a new part in the primary system test.

The "extra data" of the default app's result is populated with the relevant data, and the writer treats this result in the new part.

[ML-5767]: https://iguazio.atlassian.net/browse/ML-5767?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mlrun/db/base.py [hunk 1]
```diff
@@ -555,7 +555,7 @@ def get_model_endpoint(
         end: Optional[str] = None,
         metrics: Optional[list[str]] = None,
         features: bool = False,
-    ):
+    ) -> "mlrun.model_monitoring.model_endpoint.ModelEndpoint":
```
Comment (Eyal-Danieli) [inline mlrun/db/base.py]: any reason to keep the `"`?
```suggestion
    ) -> mlrun.model_monitoring.model_endpoint.ModelEndpoint:
```

File: mlrun/model_monitoring/applications/histogram_data_drift.py [hunk 2]
```diff
@@ -146,31 +147,41 @@ def _compute_metrics_per_feature(
...
+            extra_data={
+                EventFieldType.CURRENT_STATS: json.dumps(
+                    monitoring_context.feature_stats
+                ),
+                EventFieldType.DRIFT_MEASURES: metrics_per_feature.T.to_json(),
+                EventFieldType.DRIFT_STATUS: status.value,
+            },
         )
 
     def _get_metrics(
```
Comment (Eyal-Danieli) [inline mlrun/model_monitoring/applications/histogram_data_drift.py:204]: just a recommendation you can convert both _get_metrics & _log_json_artifact into staticmethods.
```suggestion
    @staticmethod
    def _get_metrics(
```

File: mlrun/model_monitoring/writer.py [hunk 3]
```diff
@@ -212,3 +214,22 @@ def do(self, event: _RawEvent) -> None:
...
+            and event[ResultData.RESULT_NAME]
+            == HistogramDataDriftApplicationConstants.GENERAL_RESULT_NAME
+        ):
+            endpoint_id = event[WriterEvent.ENDPOINT_ID]
+            logger.info(
+                "Updating the model endpoint with metadata specific to the histogram "
+                "data drift app",
+                endpoint_id=endpoint_id,
+            )
+            store = mlrun.model_monitoring.db.get_store_object(project=self.project)
```
Comment (Eyal-Danieli) [inline mlrun/model_monitoring/writer.py]:
```suggestion
            store = mlrun.model_monitoring.get_store_object(project=self.project)
```

**Review by Eyal-Danieli — COMMENTED** [review]
nice, few comments
in addition, even if it's not related directly to your PR, I suggest to add more description about the `HistogramDataDriftApplication`, even just saying that this application is going to calculate 3 metrics and 1 result that represents a general drift detection value.

File: mlrun/db/base.py (hunk 1 above)
Comment (jond01) [reply mlrun/db/base.py]: cd4344f

File: mlrun/model_monitoring/applications/histogram_data_drift.py (hunk 2 above)
Comment (jond01) [reply mlrun/model_monitoring/applications/histogram_data_drift.py:204]: d720750

File: mlrun/model_monitoring/writer.py (hunk 3 above)
Comment (jond01) [reply mlrun/model_monitoring/writer.py]: a8f9df9

**Comment by jond01** [comment]
> I suggest to add more description about the `HistogramDataDriftApplication`, even just saying that this application is going to calculate 3 metrics and 1 result that represents a general drift detection value.

Improved the docstrings in 1e02681.
Added the module docstrings in d5f2302.

File: mlrun/model_monitoring/applications/histogram_data_drift.py [hunk 4]
```diff
@@ -90,9 +91,27 @@ class HistogramDataDriftApplication(ModelMonitoringApplicationBaseV2):
     """
     MLRun's default data drift application for model monitoring.
 
-    The application calculates the metrics over the features' histograms.
-    Each metric is calculated over all the features, the mean is taken,
-    and the status is returned.
+    The application expects tabular numerical data, and calculates three metrics over the features' histograms.
```
Comment (Eyal-Danieli) [inline mlrun/model_monitoring/applications/histogram_data_drift.py:94]: king

**Review by Eyal-Danieli — APPROVED** [review]
LGTM

**Pull request merged**